### PR TITLE
[#130] Add code snippets for writing unit test

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/nimbl3/ios-templates/issues/130

## What happened

Add code snippets for writing unit test
 
## Insight

- Add code snippets into new Nimble Code Snippets folder
- Write new scripts to copy code snippets
 
## Proof Of Work
Screen Shot 2021-01-21 at 15 10 21

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
-# Backup old templates in case you have your custom templates
+# Copy templates
+read -r -p "Do you want to copy templates? [Y|n] " response
+if [[ $response =~ (y|yes|Y) ]];then
+  
+  # Backup old templates in case you have your custom templates
+  read -r -p "Do you want to backup your old templates? [Y|n] " response
 
-read -r -p "Do you want to backup your old templates? [Y|n] " response
+  if [[ ! $response =~ (n|no|N) ]];then
```
Comment (minhnimble) [inline install.sh:10]: why don't we just do `if [[ $response =~ (y|yes|Y) ]];then` as the same with other places. I might type `c` and it still save will probably feels a bit weird 🙏

**Review by minhnimble — APPROVED** [review]
The rest LGTM 🚀

File: install.sh (hunk 1 above)
Comment (minhnimble) [reply install.sh:10]: Or we can only accept `y|yes|Y` and `n|no|N` inputs, for other inputs we just re-ask the question? What do you think? 😁

File: Nimble Code Snippets/Unit Test/unit-test-interactor-it-trigger.codesnippet [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+<plist version="1.0">
+<dict>
+	<key>IDECodeSnippetCompletionPrefix</key>
+	<string>ittrigger</string>
+	<key>IDECodeSnippetCompletionScopes</key>
+	<array>
+		<string>All</string>
+	</array>
+	<key>IDECodeSnippetContents</key>
+	<string>it("triggers output to call &lt;#func#&gt;") {
```
Comment (minhnimble) [inline Nimble Code Snippets/Unit Test/unit-test-interactor-it-trigger.codesnippet:12]: maybe we can update to this for consistency purpose 🙏
```suggestion
	<string>it("calls output to trigger &lt;#func#&gt;") {
```

File: Nimble Code Snippets/Unit Test/unit-test-router-it-present.codesnippet [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<plist version="1.0">
+<dict>
+	<key>IDECodeSnippetCompletionPrefix</key>
+	<string>itpresent</string>
+	<key>IDECodeSnippetCompletionScopes</key>
+	<array>
+		<string>All</string>
+	</array>
+	<key>IDECodeSnippetContents</key>
+	<string>it("present &lt;#module#&gt; view") {
```
Comment (minhnimble) [inline Nimble Code Snippets/Unit Test/unit-test-router-it-present.codesnippet:12]: a small typo 🙏
```suggestion
	<string>it("presents &lt;#module#&gt; view") {
```

**Review by blyscuit — COMMENTED** [review]
Really good ideas, should we include this feature in readme too?

File: install.sh [hunk 4]
```diff
@@ -1,19 +1,59 @@
...
-if [[ ! $response =~ (n|no|N) ]];then
-  currentDate=`date +%s`
-  cp -R ~/Library/Developer/Xcode/Templates ~/Library/Developer/Xcode/Templates-Backup-${currentDate}
+  # Backup old templates in case you have your custom templates
+  confirm "Do you want to backup your old templates?"
+  response=$?
+
+  if [[ ! $response == 0 ]]; then
+    currentDate=`date +%s`
+    dest="~/Library/Developer/Xcode/Templates-Backup-${currentDate}"
```
[the remark this answers is not in the document]
Comment (markgravity) [reply install.sh:20]: @blyscuit I added a wiki for it here
https://github.com/nimblehq/ios-templates/wiki/Code-Snippets

**Review by blyscuit — APPROVED** [review]

**Pull request merged**